kernel/kernel-api/src/main/java/io/delta/kernel/internal/checkpoints/Checkpointer.java

```diff
@@ -16,29 +16,104 @@
 package io.delta.kernel.internal.checkpoints;
 
 import static io.delta.kernel.internal.DeltaErrors.wrapEngineExceptionThrowsIO;
+import static io.delta.kernel.internal.TableConfig.EXPIRED_LOG_CLEANUP_ENABLED;
+import static io.delta.kernel.internal.TableConfig.LOG_RETENTION;
+import static io.delta.kernel.internal.TableFeatures.validateWriteSupportedTable;
+import static io.delta.kernel.internal.snapshot.MetadataCleanup.cleanupExpiredLogs;
 import static io.delta.kernel.internal.util.Utils.singletonCloseableIterator;
 
 import io.delta.kernel.data.ColumnarBatch;
 import io.delta.kernel.data.Row;
 import io.delta.kernel.engine.Engine;
+import io.delta.kernel.exceptions.CheckpointAlreadyExistsException;
 import io.delta.kernel.exceptions.KernelEngineException;
+import io.delta.kernel.exceptions.TableNotFoundException;
+import io.delta.kernel.internal.SnapshotImpl;
+import io.delta.kernel.internal.actions.Metadata;
 import io.delta.kernel.internal.fs.Path;
+import io.delta.kernel.internal.replay.CreateCheckpointIterator;
 import io.delta.kernel.internal.util.*;
 import io.delta.kernel.utils.CloseableIterator;
 import io.delta.kernel.utils.FileStatus;
 import java.io.*;
+import java.nio.file.FileAlreadyExistsException;
 import java.util.*;
 import java.util.stream.Collectors;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 /** Class to load and write the {@link CheckpointMetaData} from `_last_checkpoint` file. */
 public class Checkpointer {
+
+  ////////////////////////////////
+  // Static variables / methods //
+  ////////////////////////////////
+
   private static final Logger logger = LoggerFactory.getLogger(Checkpointer.class);
 
+  private static final int READ_LAST_CHECKPOINT_FILE_MAX_RETRIES = 3;
+
   /** The name of the last checkpoint file */
   public static final String LAST_CHECKPOINT_FILE_NAME = "_last_checkpoint";
 
+  public static void checkpoint(Engine engine, Clock clock, SnapshotImpl snapshot)
```

Comment:
nearly a verbatim copy. no logic change. i change the inputs (snapshot instead of version), define tablePath and logPath and version variables, and make some minor logger improvements.